### ISSUE TITLE
Calling `saveFlags` in a callback doesn't work when Archiving a flag

### ISSUE THREAD

**Issue by KOConchobhair** [launchdarkly/c-client-sdk#90]
**Is this a support request?**
No

**Describe the bug**
I am seeing an application crash when calling `client->saveFlags()` inside a callback. I am using the `client->registerFeatureFlagListenerWithClosure("*", ...)` from #85 
I believe this line is potentially problematic
https://github.com/launchdarkly/c-client-sdk/blob/cdab1b5c08ef4a26f9e1b67a191230dad6a13823/src/flag.c#L213

**To reproduce**
`registerFeatureFlagListenerWithClosure("*", cb)
cb calls `client->saveFlags()`
Archive a flag in LD web UI

**Expected behavior**
`client->saveFlags()` should return JSON (and not return NULL)

**Logs**
```
[/tmp/tmp.fDs9DSbBhG/c-client-sdk/c-sdk-common/src/json.c, 93] LDJSONDuplicate NULL input
```

**SDK version**
2.4.7

**Language version, developer tools**
C++, gcc-10

**OS/platform**
Ubuntu 18.04 LTS

**Additional context**
I _think_ this could be similar to #87 since there a check was added to check if `!flag.deleted` before calling `LDJSONDuplicate(flag.value)` but that's just a guess on my part

**Comment by cwaldren-ld** [comment]
Hi @KOConchobhair, thanks so much for filing this report. This will certainly need to be addressed.

_Filed internally as 169687._

**Comment by cwaldren-ld** [comment]
Hi @KOConchobhair , this should be fixed by [2.5.2](https://github.com/launchdarkly/c-client-sdk/releases/tag/2.5.2). 

Please re-base the custom changes from #85 on that, and let me know if the issue is resolved.

**Comment by cwaldren-ld** [comment]
Hi @KOConchobhair , I'm closing this ticket. Please feel free to open a new one if the issue persists.
